domain-models-runtime/src/main/java/org/folio/rest/impl/JsonSchemaAPI.java

```diff
@@ -0,0 +1,168 @@
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class JsonSchemaAPI implements JsonSchema {
+
+  private static final Logger log = LoggerFactory.getLogger(JsonSchemaAPI.class);
+
+  private static final Pattern REF_MATCH_PATTERN = Pattern.compile("\\\"\\$ref\\\"\\s*:\\s*\\\"(.*?)\\\"");
+
+  @Validate
+  @Override
+  public void getJsonSchema(
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext
+  ) {
+    vertxContext.runOnContext(v -> {
+      try {
+        List<String> schemas = getSchemas();
+        asyncResultHandler.handle(
+          Future.succeededFuture(
+            GetJsonSchemaResponse.respond200WithApplicationJson(schemas)
+          )
+        );
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(
+          Future.succeededFuture(
+            GetJsonSchemaResponse.respond500WithTextPlain(e.getMessage())
+          )
+        );
+      }
+    });
+  }
+
+  @Validate
+  @Override
+  public void getJsonSchemaByName(
+    String name,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext
+  ) {
+    String okapiUrl = okapiHeaders.get("x-okapi-url");
+    vertxContext.runOnContext(v -> {
+      try {
+        JsonNode schemaNode = getSchemaByName(name, okapiUrl);
+        if(schemaNode != null) {
+          asyncResultHandler.handle(
+            Future.succeededFuture(
+              GetJsonSchemaByNameResponse.respond200WithApplicationJson(schemaNode)
+            )
+          );
+        } else {
+          String notFoundMessage = "Schema " + name + " not found";
+          asyncResultHandler.handle(
+            Future.succeededFuture(
+              GetJsonSchemaByNameResponse.respond404WithTextPlain(notFoundMessage)
+            )
+          );
+        }
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(
+          Future.succeededFuture(
+            GetJsonSchemaByNameResponse.respond500WithTextPlain(e.getMessage())
+          )
+        );
+      }
+    });
+  }
+
+  private List<String> getSchemas() throws IOException {
+    List<String> schemas = new ArrayList<>();
+    File jarFile = new File(getClass().getProtectionDomain().getCodeSource().getLocation().getPath());
+    JarFile jar = new JarFile(jarFile);
+    Enumeration<JarEntry> entries = jar.entries();
+    while(entries.hasMoreElements()) {
+      JarEntry entry = entries.nextElement();
+      String entryName = entry.getName();
+      if (entryName.startsWith("ramls/") && entryName.endsWith(".json")) {
+        String schemaPath = entryName.substring(6);
+        if(!schemaPath.contains("/")) {
```

Comment:
Does this work for ramls/instance.json?

Reply:
It does. The idea is to only check files in the ramls directory and not subdirectories.